src/nnet3/nnet-component-test.cc

```diff
@@ -54,6 +54,38 @@ void TestNnetComponentAddScale(Component *c) {
   delete c3;
 }
 
+void TestNnetComponentVectorizeUnVectorize(Component *c) {
+  UpdatableComponent *uc = dynamic_cast<UpdatableComponent*>(c);
+  if((uc==NULL) || (uc->NumParameters() == 0))
+    return;
+  UpdatableComponent *uc2 = dynamic_cast<UpdatableComponent*>(uc->Copy());
+  uc2->SetZero(false);
+  Vector<BaseFloat> params(uc2->NumParameters());
+  uc2->Vectorize(&params);
+  KALDI_ASSERT(params.Min()==0.0 && params.Sum()==0.0);
+  uc->Vectorize(&params);
+  uc2->UnVectorize(params);
+  KALDI_ASSERT(uc2->Info() == uc->Info());
+  KALDI_ASSERT(uc2->DotProduct(*uc2) == uc->DotProduct(*uc));
+  delete uc2;
```

Comment:
Actually I was disconnected before pushing and I couldn't connect for ~1 hour.
Unfortunately my connection to CLSP grid is very unreliable and slow at some times. I'm not sure why.